### PR TITLE
[IGNORE] Refactor to have a single PrometheusDatasourceSpec

### PR DESCRIPTION
**Pull request by AntoineThebaud**
# Description

Refactor to have a single PrometheusDatasourceSpec type in the frontend code.

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

### CODE REVIEW

File: ui/prometheus-plugin/src/plugins/types.ts [hunk 1]
```diff
@@ -11,8 +11,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { DurationString, RequestHeaders } from '@perses-dev/core';
 import { PrometheusDatasourceSelector } from '../model';
 
+export const DEFAULT_SCRAPE_INTERVAL: DurationString = '1m';
```
Comment (AntoineThebaud) [inline ui/prometheus-plugin/src/plugins/types.ts:17]: moved from previous `PrometheusDatasourceEditor/types.ts` file

File: ui/prometheus-plugin/src/plugins/prometheus-datasource.tsx [hunk 2]
```diff
@@ -11,25 +11,17 @@
...
+import { PrometheusDatasourceSpec } from './types';
+import { PrometheusDatasourceEditor } from './PrometheusDatasourceEditor';
 
 /**
  * Creates a PrometheusClient for a specific datasource spec.
  */
 const createClient: DatasourcePlugin<PrometheusDatasourceSpec, PrometheusClient>['createClient'] = (spec, options) => {
-  const { directUrl, headers: specHeaders } = spec;
+  const { directUrl, proxy } = spec;
   const { proxyUrl } = options;
```
Comment (AntoineThebaud) [inline ui/prometheus-plugin/src/plugins/prometheus-datasource.tsx:25]: FYI `proxyUrl` is the "proxified" url ready to be used for the query calls (see buildProxyUrl func), it is not equal to `proxy.spec.url` here, hence why this part is not touched.

File: ui/prometheus-plugin/src/plugins/types.ts [hunk 3]
```diff
@@ -11,8 +11,41 @@
...
+
+// TODO: unify this one with the other PrometheusDatasourceSpec used for datasource store manipulation
```
Comment (Gladorme) [inline ui/prometheus-plugin/src/plugins/types.ts:19]: Can be removed now 😄

**Review by Gladorme — APPROVED** [review]
LGTM, just a little todo to remove 👍

File: ui/prometheus-plugin/src/plugins/types.ts (hunk 3 above)
Comment (AntoineThebaud) [reply ui/prometheus-plugin/src/plugins/types.ts:19]: oups! thanks for spotting this

**Pull request merged**